Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectBuilder/MSBuildProjectBuilder.cs

```diff
@@ -155,20 +157,28 @@ public static bool BuildProjects(this IReadOnlyCollection<MSBuildProjectReferenc
                     {
                         int completedProjects = 0;
                         string progressMessage = string.Empty;
+                        Match adoAuthenticationMatch = null;
                         var cancellationTokenSource = new CancellationTokenSource();
 
                         DisplayProgress();
 
                         Task<bool> buildTask = MSBuildProjectBuilder.BuildProjectsAsync(
                             msBuildProjectReferences,
-                            $"{arguments} -v:minimal",
+                            $"{arguments} -v:minimal -p:NuGetInteractive=true",
                             new DelegateProgress<(int completedProjects, (string progressMessage, ProgressMessageType progressMessageType) progressUpdate)>(report =>
                             {
                                 if (report.progressUpdate.progressMessageType != ProgressMessageType.Information)
                                 {
                                     MSBuildProjectBuilder.LogProgressMessage(report.progressUpdate.progressMessage, report.progressUpdate.progressMessageType);
                                 }
+
                                 (completedProjects, progressMessage) = (report.completedProjects, report.progressUpdate.progressMessage);
+
+                                // Check whether the build is blocked on Azure DevOps package feed authentication
+                                if (MSBuildProjectBuilder.adoAuthenticationFormat.Match(progressMessage) is Match candidateAdoAuthenticationMatch && candidateAdoAuthenticationMatch.Success)
```

Comment:
>MSBuildProjectBuilder.adoAuthenticationFormat.Match(progressMessage) is Match candidateAdoAuthenticationMatch && candidateAdoAuthenticationMatch.Success [](start = 36, length = 152)

I feel this is using just as much text as:
```
Match candidateAdoAuthenticationMatch = MSBuildProjectBuilder.adoAuthenticationFormat.Match(progressMessage);
if (candidateAdoAuthenticationMatch.Success)
```

but it comes across as being harder to read.

Reply:
Addressed with some minor changes.